test/functional/mining_basic.py

```diff
@@ -49,6 +51,9 @@ def set_test_params(self):
         self.setup_clean_chain = True
         self.supports_cli = False
 
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
```

Comment:
This is ok for now, but it'd be nice to rework this test so that it doesn't require a wallet.

We could also update the test to verify that the witness commitment commits to the wtxids of the transactions in the block rather than the txids. Currently the block being tested contains only the coinbase transaction and one transaction that spends from a P2PKH output, so the txid is the same as wtxid. It could be updated to include transactions that spend segwit outputs so that the txid and wtxid are different.

Reply:
I can do those in a follow-up PR. Can you tell me more about how I could remove the wallet dependency? My understanding is that to have `getblocktemplate` include a transaction that the witness root commits to, I need a valid transaction in the mempool, and to create that valid tx, I need a wallet. I feel like I am missing something about the test framework perhaps?

Reply:
@dhruv: Luckily we have a minimalistic wallet implementation for the functional test framework (first introduced by MarcoFalke last year, see #19800). Look for modules importing `MiniWallet` for example usage, see also issue #20078. Also, feel free to tag me in your follow-up PR, will be happy to review.

Reply:
Agree